### PR TITLE
Auto PR: Regenerating based on (b1e38bdd82d23036cec64ebcc0b64d7c8cb8930a)

### LINKED ISSUE #21629: Support for AKS trusted access
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AKS recently offers trusted access. Trusted Access is required for Azure Backup and AKS Private Clusters

### New or Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster_trusted_access_role_binding

### Potential Terraform Configuration

```hcl
resource "azurerm_kubernetes_cluster_trusted_access_role_binding" "this" {
  name           = "bindingName"
  clustr_id     = azurerm_kubernetes_cluster.example.id
  source_resource_id = "/subscriptions/0000/resourceGroups/myResourceGroup/providers/Microsoft.Demo/samples"
  roles = ["Microsoft.Demo/samples/reader","Microsoft.Demo/samples/writer"]
}
```

### References

* https://learn.microsoft.com/en-us/azure/aks/trusted-access-feature

### PR DESCRIPTION
**Pull request by hc-github-team-tf-azure**
Regenerating the Terraform Provider based on the latest changes

Closes #21629

### CODE REVIEW

**Comment by stephybun** [comment]
Tests are failing - may need to regenerate this with an older API version

File: website/docs/r/kubernetes_cluster_trusted_access_role_binding.html.markdown [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+```hcl
+resource "azurerm_application_insights" "example" {
+  name                = "example"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  application_type    = "example-value"
+}
+data "azurerm_client_config" "test" {}
+resource "azurerm_key_vault" "example" {
```
Comment (katbyte) [inline website/docs/r/kubernetes_cluster_trusted_access_role_binding.html.markdown:27]: the formatting here is off, should be some blank lines between resources?

File: website/docs/r/kubernetes_cluster_trusted_access_role_binding.html.markdown [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+description: |-
+  Manages a Kubernetes Cluster Trusted Access Role Binding.
+---
+
+<!-- Note: This documentation is generated. Any manual changes will be overwritten -->
+
+# azurerm_kubernetes_cluster_trusted_access_role_binding
+
+Manages a Kubernetes Cluster Trusted Access Role Binding
+~> **Note:** This Resource is in **Preview** to use this you must be opted into the Preview. You can do this by running `az feature register --namespace Microsoft.ContainerService --name TrustedAccessPreview` and then `az provider register -n Microsoft.ContainerService`
```
Comment (katbyte) [inline website/docs/r/kubernetes_cluster_trusted_access_role_binding.html.markdown:14]: a blank limne before the note?

**Review by katbyte — APPROVED** [review]
LGTM aside from some minor formatting comment

**Comment by tombuildsstuff** [comment]
Closing in favour of #22647

**Pull request closed without merging**